test/distributed/test_functional_api.py

```diff
@@ -412,6 +412,13 @@ def setUp(self):
         super().setUp()
         self._spawn_threads()
 
+    def tearDown(self):
+        super().tearDown()
+
+        # race condition with threads causes is_fx_tracing flag to be set incorrectly.
+        torch.fx._symbolic_trace._is_fx_tracing_flag = False
```

Comment:
I have been hitting race conditions recently with multi-threaded process group when patching functions or affecting global state (some fixed by adding `dist.barrier()`s: https://github.com/pytorch/pytorch/pull/123754).

My question here is would we have to end up adding patches into this `tearDown` for _any_ global state mutation that could happen?

Reply:
Mean, the best thing here would be to put a fix into fx so it would track these variables using thread local state rather than a global

I'm not sure how much global state there is -- using threads for things that use globals are inherently dangerous

I did consider switching this to use processes